knfc/knfconverter.js

```diff
@@ -0,0 +1,220 @@
+// Generated JS from Java: Node -----
+function Node() {
+    this.left = null;
+    this.right = null;
+    this.parent = null;
+    this.s = null;
+    this.id = null;
+ 
+    if (arguments.length == 0) {
+       jv_Object.call(this);
+    this._NodeInit();
+    }
+    else if (arguments.length == 1) {
+       var s = arguments[0];
+       jv_Object.call(this);
+    this._NodeInit();
+       this.s = s;
+    }}
+ 
+ var Node_c = sc_newClass("Node", Node, jv_Object, null);
+ 
+ Node_c.toString = function ()  {
+    if (this.hasOwnProperty("$protoName")) {
+       return jv_Class_c.toString.apply(this, arguments);
+    }
+    return "[" + (this.s !== null ? this.s : "") + (this.id !== null ? " id=" + this.id : "") + (this.left !== null ? (" : " + this.left.toString() + (this.right !== null ? ("," + this.right.toString()) : "")) : "") + "]";
+ };
+ Node_c.setIDRec = function (id)  {
```

Comment:
Man kann so zwar Funktionen definieren, jedoch finde ich des unschön.
Schau dir mal hier an, wie man JS Klassen erstellt, ist gar nicht so anders als in Java
https://javascript.info/class

Reply:
Du kannst dir auch in unserem code anschauen, wie Dorian des gemacht hat.
https://github.com/keinenclue/keinenclue.github.io/blob/main/pcc/permutationCalculator.js

Reply:
Ah und splitte bitte die beiden Klassen auf, mach eine Datei für Node und eine Datei für die andere Klasse